# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/browser-siteIdentity.js
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 2
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e browser/base/content/browser-siteIdentity.js
--- a/browser/base/content/browser-siteIdentity.js	Mon Oct 21 19:33:02 2019 +0000
+++ b/browser/base/content/browser-siteIdentity.js	Mon Oct 21 19:49:01 2019 +0000
@@ -677,10 +677,9 @@
    */
   _hasCustomRoot() {
     let issuerCert = null;
-    // Walk the whole chain to get the last cert.
-    // eslint-disable-next-line no-empty
-    for (issuerCert of this._secInfo.succeededCertChain.getEnumerator()) {
-    }
+    issuerCert = this._secInfo.succeededCertChain[
+      this.secInfo.succeededCertChain.length - 1
+    ];
 
     return !issuerCert.isBuiltInRoot;
   },